typing/printtyp.ml

```diff
@@ -242,6 +269,18 @@ let with_hidden ids f =
   let updated = List.fold_left update !bound_in_recursion ids in
   protect_refs [ R(bound_in_recursion, updated )] f
 
+let human_id id index =
+  if index = 0 then
+    Ident.name id
+  else
+  (* We print the most k-time overshadowed identifier as `name/(k+1)`.
```

Comment:
I'm not sure what "most k-time overshadowed" means.

I think you could have a shorter explanation anyway: humans are more comfortable with 1-indexing instead of 0-indexing so we do that, and we show `foo` instead of `foo/1`.

(I'm not sure I like this choice to be honest. It is natural to think of `foo/n` as "the (1-indexed) n-th shadowed definition", so `foo/1` is the first shadowed definition so it is different from `foo`.)

Reply:
```
The identifier with index [k] is the (k+1)-th most recent identifiers in
the printing environment. We print it as [name/(k+1)], except for [k=0]
which is printed as [name] rather than [name/1].
```
?

Reply:
'most recent identifier' rather than 'most recent identifiers'